### PR TITLE
feat: add more get/list commands

### PR DESCRIPTION
**Pull request by Lanture1064**
#2

### CODE REVIEW

File: pkg/federation/get.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+					APIVersion: "v1",
+				},
+				ListMeta: v1.ListMeta{},
+			}
+
+			var obj runtime.Object
+
+			//
+			if len(args) == 0 {
+				feds, err := cli.Resource(schema.GroupVersionResource{Group: common.IBPGroup, Version: common.IBPVersion, Resource: common.FederationResource}).List(context.TODO(), v1.ListOptions{
```
Comment (bjwswang) [inline pkg/federation/get.go:70]: `List` seems not working for `Federations`. Have you tried this command before make this PR?

File: pkg/federation/get.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubectl/pkg/cmd/get"
+)
+
+func NewFedGetCmd(option common.Options) *cobra.Command {
```
Comment (bjwswang) [inline pkg/federation/get.go:35]: 1. get all federations that current user joins
```shell
bc-cli get feds
```
2. get a specific federation
```shell
bc-cli get fed {fed-name}
```
support:
- `-oyaml`
- `-ojson`

3. get  allfederation that belongs to a specific org
```shell
bc-cli get feds  --with-org {org_name}
```

File: pkg/federation/get.go (hunk 1 above)
Comment (0xff-dev) [reply pkg/federation/get.go]: Users only have get access to the federation, not list access. So we need to get a certain organization, get the feds the organization has joined, and get the list of feds.

**Review by bjwswang — APPROVED** [review]

File: pkg/federation/get.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+			orgName, _ := cmd.Flags().GetString("with-org")
+
+			if orgName == "" {
+				if len(args) == 0 {
+					// get all feds of all orgs
+					orgs, err := cli.Resource(schema.GroupVersionResource{
+						Group:    common.IBPGroup,
+						Version:  common.IBPVersion,
+						Resource: common.OrganizationResource,
+					}).List(context.TODO(), v1.ListOptions{LabelSelector: "user=account"}) // FIXME: Selector
```
Comment (0xff-dev) [inline pkg/federation/get.go:71]: Do we need to add `LabelSelector`? According to my tests, it is possible to get all the organization information, even if the current user is not the creator of the organization.
@bjwswang

Comment (bjwswang) [reply pkg/federation/get.go]: @0xff-dev @Lanture1064 well... yes.. we need to list the organizations in which this user is the client/admin

Comment (bjwswang) [reply pkg/federation/get.go]: @0xff-dev please help with @Lanture1064

Comment (0xff-dev) [reply pkg/federation/get.go]: @Lanture1064  Merge  the code of`connProfile`, which will have `username`  in viper's configuration, the organization information is obtained from the `annotations["bestchains"]` field of the user. Then get the information about the federation to which each organization belongs by get org.

```json
{
  "list": {
    "org1": {},
    "org2": {},
  }
}
```

**Comment by bjwswang** [comment]
@0xff-dev resolve the conversations if got fixed

File: pkg/federation/get.go [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+								Version:  common.IBPVersion,
+								Resource: common.OrganizationResource,
+							}).Get(context.TODO(), org, v1.GetOptions{})
+							if err != nil {
+								fmt.Fprintln(option.ErrOut, err)
+								return err
+							}
+
+							feds, _, _ := unstructured.NestedStringSlice(orgObj.Object, "status", "federation")
+
```
Comment (0xff-dev) [inline pkg/federation/get.go:101]: `federations`

File: pkg/federation/get.go [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+					Version:  common.IBPVersion,
+					Resource: common.OrganizationResource,
+				}).Get(context.TODO(), orgName, v1.GetOptions{})
+				if err != nil {
+					fmt.Fprintln(option.ErrOut, err)
+					return err
+				}
+
+				feds, _, _ := unstructured.NestedStringSlice(org.Object, "status", "federation")
+
```
Comment (0xff-dev) [inline pkg/federation/get.go:148]: `federations`

File: pkg/federation/get.go [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+limitations under the License.
+*/
+
+package federation
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/viper"
```
Comment (0xff-dev) [inline pkg/federation/get.go:23]: import format.

File: pkg/federation/get.go [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+
+			orgName, _ := cmd.Flags().GetString("with-org")
+
+			if orgName == "" {
+				if len(args) == 0 {
+					// get all feds of all orgs
+					// - get orgList from username
+					username := viper.GetString("auth.username")
+					u, err := cli.Resource(schema.GroupVersionResource{
+						Group:    "iam.tenxcloud.com",
```
Comment (0xff-dev) [inline pkg/federation/get.go:74]: can be defined as constants in common package.

**Comment by 0xff-dev** [comment]
/lgtm  Solve the last two conversations, I think we can merge the code.
```shell
➜  bc-cli git:(main) ✗ ./main get fed
NAME    AGE
proof   35d
test    35d
2346    31d
➜  bc-cli git:(main) ✗ ./main get fed --with-org=tenxcloud
NAME    AGE
proof   35d
test    35d
2346    31d
➜  bc-cli git:(main) ✗ ./main get fed proof               
NAME    AGE
proof   35d
➜  bc-cli git:(main) ✗ ./main get fed proof --show-labels
NAME    AGE   LABELS
proof   35d   bestchains.federation.initiator=tenxcloud
```

**Review by bjwswang — APPROVED** [review]
/LGTM
@Lanture1064 @0xff-dev Awesome!

**Pull request merged**